math/fibonacci_matrix_exponentiation.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file fibonacci_matrix_exponentiation.cpp
+ * @brief Computes N^th Fibonacci number given as
```

Comment:
Not true. This program computes the "N^th Fibonacci number in modulo M" which is a bit different.